File: tools/reposted_by.py
#!/usr/bin/env python3
"""List the accounts that reposted a specific post (walks the engagement graph).

    python3 reposted_by.py at://did:plc:.../app.bsky.feed.post/xyz --limit 50
"""
import argparse
import sys

from actor_index import print_actors
from bsky import BskyError, get


def main() -> None:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("uri", help="at:// URI of the post")
    parser.add_argument("--limit", type=int, default=50)
    args = parser.parse_args()
    data = get("app.bsky.feed.getRepostedBy", {"uri": args.uri, "limit": args.limit})
    print_actors(data.get("repostedBy", []))


if __name__ == "__main__":
    try:
        main()
    except BskyError as e:
        print(f"reposts fetch failed: {e}", file=sys.stderr)
        sys.exit(1)

File: agent/.opencode/tools/reposted-by.ts
import { tool } from "@opencode-ai/plugin"
import { $ } from "bun"

// Thin wrapper around tools/reposted_by.py — located via the git worktree.
export default tool({
  description:
    "List the accounts that reposted a specific post (walks the engagement " +
    "graph). Pass the post's at:// URI. Returns one line per account: " +
    "`@handle (Name) — did · bio`. Reposts are a strong signal of who " +
    "amplified a post — useful for tracing how something spread.",
  args: {
    uri: tool.schema.string().describe("at:// URI of the post"),
    limit: tool.schema.number().describe("Max accounts to return").default(50),
  },
  async execute(args, context) {
    const script = `${context.worktree}/tools/reposted_by.py`
    const limit = Number.isFinite(args.limit) ? Math.trunc(args.limit as number) : 50
    return await $`python3 ${script} ${args.uri} --limit ${String(limit)}`.env({ ...process.env, ACTING_SESSION: context.sessionID }).text()
  },
})
